Added AES-128 encryption to the generated files

kavuri opened the pull request:
Added switch -e to encrypt the segments. Followed the instructions
from here: http://www.barryodonovan.com/2010/08/30/http-streaming-with-encryption-under-linux-2

Signed-off-by: Sateesh Kavuri <sateesh.kavuri@gmail.com>

bentasker commented:
Hi,

Thanks, could you make one minor change and then I'll merge:

As the change won't work with Linear video (the stream would need to end before the encryption process could start) could you amend the encrypt check to also check whether it's a live stream?

I.e.
    if [ "$ENCRYPT" == "1" ]
    then

Becomes

    if [ "$ENCRYPT" == "1" ] && [ "$LIVE_STREAM" == "0" ]
    then

Other than that, looks good thanks.

kavuri commented:
Added condition for live stream check and updated to the same pull request

Thanks,
Sateesh

On Fri, Jul 3, 2015 at 9:23 PM, Ben Tasker <notifications@github.com> wrote:

> Hi,
>
> Thanks, could you make one minor change and then I'll merge:
>
> As the change won't work with Linear video (the stream would need to end
> before the encryption process could start) could you amend the encrypt
> check to also check whether it's a live stream?
>
> I.e.
> if [ "$ENCRYPT" == "1" ]
> then
>
> Becomes
>
> if [ "$ENCRYPT" == "1" ] && [ "$LIVE_STREAM" == "0" ]
> then
>
> Other than that, looks good thanks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bentasker/HLS-Stream-Creator/pull/7#issuecomment-118380025>
> .
>

bentasker commented:
Awesome thanks, looks good to me so merging
